### PR TITLE
close #6 - change map to unordered

### LINKED ISSUE #6: Is map type unordered?
Spec suggests encoding json as an example and json is unordered, however it doesn't explicitly states this.

Ordered hash maps would complicate implementation for some languages, like go(map) or python(dict).

copy from https://github.com/antirez/RESP3/issues/30

### PR DESCRIPTION
**Pull request by leibale**
see:
* https://github.com/redis/redis/pull/8391
* https://github.com/redis/redis/pull/8504

### CODE REVIEW

**Comment by itamarhaber** [comment]
@redis/core-team please approve this change to the spec.

**Review by oranagra — APPROVED** [review]
i approve.
we kinda already dealt with this here: https://github.com/redis/redis/pull/8391
realizing that we may want to stay away from that and never rely on map being ordered.

**Comment by leibale** [comment]
Should we add it to the "version history" section at the top?

**Review by zuiderkwast — APPROVED** [review]
Merge this?

Regarding version history, if we merge more PRs at the same time, we can have one version history bullet combining the changes.

**Comment by oranagra** [comment]
i added two links in the top about the places were we realized where map should be ordered (included a breaking change in 6.2).
@zuiderkwast please take a quick look and i'll merge it.

**Review by itamarhaber — APPROVED** [review]
LGTM

File: protocol/RESP3.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Versions history:
                    was made.
 * 1.3, 11 Mar 2019, Streamed strings and streamed aggregated types.
 * 1.4, 8 Dec 2022, Normalize NaN to a single representation "nan" and forbid "-nan" (Effective since Redis 7.2).
+* 1.4, 5 Jan 2023, Mark the map type as being unordered rather than ordered (Effective since Redis 6.2)
```
Comment (zuiderkwast) [inline protocol/RESP3.md:12]: @oranagra Sorry I was out today.

I see that now we have two 1.4 releases with different dates...

Comment (oranagra) [reply protocol/RESP3.md:12]: thanks. i'll fix.

**Pull request merged**